cairo/tests/ethereum/cancun/vm/instructions/test_system.py

```diff
@@ -31,8 +33,45 @@ def system_strategy(draw):
     return evm
 
 
+@composite
+def beneficiary_from_state(draw):
+    """Generate test cases for selfdestruct instruction.
+
+    This strategy generates an EVM instance with a beneficiary address on the stack:
+    - 80% chance: address from state (existing accounts)
+    - 20% chance: random address
+    - Always pushes the address to the stack
+    """
+    evm = draw(
+        EvmBuilder()
+        .with_env()
+        .with_message()
+        .with_stack()
+        .with_gas_left()
+        .with_running()
+        .with_accessed_addresses()
+        .with_accessed_storage_keys()
+        .with_accounts_to_delete()
+        .with_touched_accounts()
```

Comment:
you can add with created_accounts too for extra cases (the case where is_created != 0)

Reply:
added in with_env